distributed/core.py

```diff
@@ -663,15 +663,25 @@ async def send_recv(comm, reply=True, serializers=None, deserializers=None, **kw
             response = await comm.read(deserializers=deserializers)
         else:
             response = None
-    except OSError:
+    # Note: this is asyncio.TimeoutError, not the builtin TimeoutError which is a
+    # subclass of OSError
+    except (TimeoutError, OSError):
         # On communication errors, we should simply close the communication
+        # This includes CommClosedError and socket timeouts
         force_close = True
         raise
+    except (CancelledError, asyncio.CancelledError):
```

Comment:
Same as above, I don't have evidence that ``concurrent.futures.CancelledError`` is ever raised.

Reply:
Ok, after reading the tornado code, there's no part of our async code that should ever hit `concurrent.futures.CancelledError`. Maybe this would happen for older versions of tornado, but not any version we're currently using.

Reply:
@jcrist same question as above

Reply:
removed ``concurrent.futures.CancelledError``